context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +164,19 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  /**
+   * Time source representing nanoseconds since fixed but arbitrary point in time.
+   *
+   * <p>This is public for testing purposes.
+   */
+  public interface DeadlineTicker {
```

Comment:
?? I don't see what stuttering goes away with the `DeadlineTicker` name when using Guava. And that isn't how it would look. It would look like:

```java
class MyTicker extends Ticker implements Deadline.Ticker {}
```

Vs your rename:
```java
import io.grpc.Deadline.DeadlineTicker;

class MyTicker extends Ticker implements DeadlineTicker {}
```

I don't understand how it limits scope, but I do see the autocomplete argument.